packages/monaco/src/index.tsx

```diff
@@ -90,87 +86,122 @@ export function CodePanel(props: CodePanelProps) {
       });
     }
   };
+  const hasFailingTest = tsErrors?.some((e) => e.length) ?? false;
 
   return (
     <>
-      <div className="sticky top-0 flex h-[40px] items-center justify-end gap-4 border-b border-zinc-300 px-3 py-2 dark:border-zinc-700 dark:bg-[#1e1e1e]">
+      <div className="sticky top-0 flex h-[40px] shrink-0 items-center justify-end gap-4 border-b border-zinc-300 px-3 py-2 dark:border-zinc-700 dark:bg-[#1e1e1e]">
         {props.settingsElement}
       </div>
-      <div className="w-full flex-1">
-        <SplitEditor
-          monaco={monacoInstance}
-          tests={props.challenge.tests}
-          userCode={code}
-          onMount={{
-            tests: (editor) => {
-              setTestEditorState(editor);
-              setTsErrors([[], [], [], []]);
-            },
-            user: async (editor, monaco) => {
-              setMonacoInstance(monaco);
-
-              monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
-                ...monaco.languages.typescript.typescriptDefaults.getCompilerOptions(),
-                strict: true,
-                target: monaco.languages.typescript.ScriptTarget.ESNext,
-                strictNullChecks: true,
-              });
-
-              monaco.languages.registerDocumentFormattingEditProvider(
-                'typescript',
-                PrettierFormatProvider,
-              );
-
-              setUserEditorState(editor);
-
-              const model = editor.getModel();
-
-              if (!model) {
-                throw new Error();
-              }
-
-              const ts = await (await monaco.languages.typescript.getTypeScriptWorker())(model.uri);
-
-              monaco.languages.registerInlayHintsProvider(
-                'typescript',
-                createTwoslashInlayProvider(monaco, ts),
-              );
-            },
-          }}
-          onChange={{
-            user: async (code) => {
-              if (!monacoInstance) return null;
-              setCode(code ?? '');
-              setLocalStorageCode(code ?? '');
-              const getTsWorker = await monacoInstance.languages.typescript.getTypeScriptWorker();
-
-              const mm = monacoInstance.editor.getModel(monacoInstance.Uri.parse(TESTS_PATH));
-              if (!mm) return null;
-
-              const tsWorker = await getTsWorker(mm.uri);
-
+      <SplitEditor
+        userEditorState={userEditorState}
+        monaco={monacoInstance}
+        expandTestPanel={isTestPanelExpanded}
+        tests={props.challenge.tests}
+        userCode={code}
+        onMount={{
+          tests: async (editor, monaco) => {
+            const model = editor.getModel();
+
+            if (!model) {
+              throw new Error();
```

Comment:
question: what is handling this error? If it's some ErrorBoundary, should we specify the error message?

Reply:
not sure need to check. iirc the editor just stops working and it probably goes to sentry will confirm later!